### PR TITLE
CLN: Replaced package list with find_packages

### LINKED ISSUE #18535: Leveraging find_packages in setup.py
Right now when creating new folders we have to also list them as packages in the setup.py file. Is there any reason why we have not considered using find_packages as suggested by the [Python Packaging User Guide](https://packaging.python.org/tutorials/distributing-packages/#packages) to take care of that? 

Happy to submit a PR just not an expert on packaging, so wanted to discuss what if any caveats are out there first before making any changes

### PR DESCRIPTION
**Pull request by WillAyd**
- [X] closes #18535 
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

As far as local testing goes, I built source distributions using py27 and py36. Compared to HEAD, the generated distributions were practically identical. Technically they were off by a size of around 100 bytes, but I was getting that same variation when generating clean built distributions just on HEAD alone. Running a recursive diff showed some slight differences in the ``.so`` files, but given I saw that variation without making any changes I think it is OK.

### CODE REVIEW

File: doc/source/whatsnew/v0.22.0.txt [hunk 1]
```diff
@@ -262,4 +262,5 @@ Other
 - Fixed a bug where creating a Series from an array that contains both tz-naive and tz-aware values will result in a Series whose dtype is tz-aware instead of object (:issue:`16406`)
 - Fixed construction of a :class:`Series` from a ``dict`` containing ``NaN`` as key (:issue:`18480`)
 - Adding a ``Period`` object to a ``datetime`` or ``Timestamp`` object will now correctly raise a ``TypeError`` (:issue:`17983`)
+- Refactored ``setup.py`` to use ``find_packages`` instead of explicitly listing out all subpackages (:issue:`18535`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: can you move to the Other API changes (where we put cython); you can actually move those 2 to a new sub-section, maybe Development Requirements or something (after Other API changes).

File: ci/requirements_dev.txt [hunk 2]
```diff
@@ -6,5 +6,5 @@ moto
 pytest>=3.1
 python-dateutil>=2.5.0
 pytz
-setuptools
-sphinx
\ No newline at end of file
+setuptools>=3.3
```
Comment (jreback) [inline ci/requirements_dev.txt]: this is fine as this is a pretty old version. can you add this to the conda recipe as well.

File: ci/requirements_dev.txt [hunk 3]
```diff
@@ -6,5 +6,5 @@ moto
 pytest>=3.1
 python-dateutil>=2.5.0
```
Comment (jreback) [inline ci/requirements_dev.txt]: can you update in doc/source/install.rst as well

File: setup.py [hunk 4]
```diff
@@ -712,65 +712,7 @@ def pxd(name):
...
-                'pandas.tests.tseries',
-                'pandas.tests.tseries.offsets',
-                'pandas.tests.plotting',
-                'pandas.tests.tools',
-                'pandas.tests.util',
-                'pandas.tools',
-                'pandas.tseries',
-                ],
+      packages=find_packages(include=['pandas', 'pandas.*']),
       package_data={'pandas.tests': ['data/*.csv'],
```
Comment (jreback) [inline setup.py]: I think there is also a utility to find various package data as well?

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. some doc changes.

File: setup.py (hunk 4 above)
Comment (WillAyd) [reply setup.py]: Below is what I could find in the setuptools documentation on options to include package data. I went with the second suggestion and just included a wildcard entry for all of the extensions currently in use.

http://setuptools.readthedocs.io/en/latest/setuptools.html#including-data-files

File: ci/requirements_dev.txt (hunk 2 above)
Comment (jreback) [reply ci/requirements_dev.txt]: can you update this?

**Review by jreback — CHANGES_REQUESTED** [review]
https://travis-ci.org/pandas-dev/pandas/jobs/311618937 is failing. In fact this is the key test as it pip installs and then tests from not-the-build directory, so it tests whether all of the data is correctly included in setup. ping when green.

File: ci/requirements_dev.txt (hunk 2 above)
Comment (WillAyd) [reply ci/requirements_dev.txt]: What are you looking to have updated?

Comment (jreback) [reply ci/requirements_dev.txt]: pandas/conda.recipe/meta.yml

**Comment by jreback** [comment]
can you rebase this once again.

**Review by jreback — APPROVED** [review]

**Comment by WillAyd** [comment]
The AppVeyor failure is for a plotting test and does not appear to be related to any change made here

**Comment by jreback** [comment]
thanks @WillAyd

**Pull request merged**